internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -889,13 +892,9 @@ func (r *oauthClientResource) ValidateConfig(ctx context.Context, req resource.V
 		if grantTypeVal == "CLIENT_CREDENTIALS" {
 			if clientAuthDefined {
 				clientAuthType := clientAuthAttributes["type"].(types.String).ValueString()
-				clientAuthSecret := clientAuthAttributes["secret"].(types.String).ValueString()
 				if clientAuthType != "NONE" && clientAuthType != "SECRET" {
 					resp.Diagnostics.AddError("client_auth.type must be set to \"SECRET\" when \"CLIENT_CREDENTIALS\" is included in grant_types.", "")
 				}
-				if clientAuthSecret == "" {
```

Comment:
What was the issue here?

Reply:
I'm reworking this now, this is incorrect